File: WEEK6/Js/bj5_11051.js
const fs = require('fs');
const [N, K] = fs.readFileSync('/dev/stdin').toString().trim().split(' ').map(Number);

const dp = Array.from({ length: N + 1 }, () => new Int32Array(N + 1));

for (let i = 0; i <= N; i++) {
    for (let j = 0; j <= i; j++) {
        if (j === 0 || j === i) {
            dp[i][j] = 1;
        } else {
            dp[i][j] = (dp[i - 1][j - 1] + dp[i - 1][j]) % 10007;
        }
    }
}

console.log(dp[N][K]);

File: WEEK6/python/bj5_11051.py
import sys

input = sys.stdin.readline

n, k = map(int, input().split())
dp = [[0] * (n + 1) for _ in range(n + 1)]

for i in range(n + 1):
    for j in range(i + 1):
        if j == 0 or j == i:
            dp[i][j] = 1
        else:
            dp[i][j] = (dp[i - 1][j - 1] + dp[i - 1][j]) % 10007

print(dp[n][k])
